x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/details/index.tsx

```diff
@@ -416,39 +411,21 @@ const RuleDetailsPageComponent: React.FC<DetectionEngineComponentProps> = ({
 
   const alertDefaultFilters = useMemo(
     () => [
-      ...buildAlertsRuleIdFilter(ruleId),
-      ...(ruleRegistryEnabled
-        ? [
-            ...buildShowBuildingBlockFilterRuleRegistry(showBuildingBlockAlerts), // TODO: Once we are past experimental phase this code should be removed
-            ...buildAlertStatusFilterRuleRegistry(filterGroup),
-          ]
-        : [
-            ...buildShowBuildingBlockFilter(showBuildingBlockAlerts),
-            ...buildAlertStatusFilter(filterGroup),
-          ]),
+      ...buildAlertsRuleIdFilter(rule?.rule_id ?? ''),
```

Comment:
Would be nice to refactor this component so that these checks aren't necessary. Right now we're fetching the rule via `useRuleWithFallback` and have the page in a loading state while `rule == null`, and all these filters that are being generated aren't needed till the rule returns anyway.